src/from_substrait.cpp

```diff
@@ -636,7 +638,33 @@ shared_ptr<Relation> SubstraitToDuckDB::TransformReadOp(const substrait::Rel &so
 		} else {
 			scan = GetValuesExpression(sget.virtual_table().expressions());
 		}
-
+	} else if (sget.has_iceberg_table()) {
+		if (sget.iceberg_table().direct().metadata_uri().empty()) {
+			throw InvalidInputException("Metadata file missing in iceberg table read in substrait");
+		}
+		string name = "iceberg_" + StringUtil::GenerateRandomName();
+		named_parameter_map_t named_parameters({});
+		vector<Value> parameters {sget.iceberg_table().direct().metadata_uri()};
+		if (sget.iceberg_table().direct().has_snapshot_id()) {
+			auto str = sget.iceberg_table().direct().snapshot_id();
+			int64_t snapshot_id = strtoimax(str.c_str(), nullptr, 10);
+			if (snapshot_id == 0) {
```

Comment:
should this be <1 or are negative numbers allowed?

Reply:
From what I can tell negative numbers are not allowed so I've added that restriction.  I've also added an additional check for max_int which could also indicate a numeric conversion issue.